File: solutions/typescript/01-yy2/code/app/main.ts
import OpenAI from "openai";

async function main() {
    const [,, flag, prompt] = process.argv;
    const apiKey = process.env.OPENROUTER_API_KEY;
    const baseURL = process.env.OPENROUTER_BASE_URL;

    if (!apiKey) {
        console.error("error: OPENROUTER_API_KEY environment variable is not set");
        process.exit(1);
    }
    if (!baseURL) {
        console.error("error: OPENROUTER_BASE_URL environment variable is not set");
        process.exit(1);
    }
    if (flag !== "-p" || !prompt) {
        console.error("error: -p flag is required");
        process.exit(1);
    }

    const client = new OpenAI({
        apiKey: apiKey,
        baseURL: baseURL,
    });
    
    const response = await client.chat.completions.create({
        model: "anthropic/claude-haiku-4.5",
        messages: [{ role: "user", content: prompt }],
    });

    if (!response.choices || response.choices.length === 0) {
        console.error("error: no choices in response");
        process.exit(1);
    }

    console.log(response.choices[0].message.content);
}

main().catch(err => {
    console.error(`error: ${err.message}`);
    process.exit(1);
});

File: solutions/python/01-yy2/code/app/main.py
import argparse, os, sys
from openai import OpenAI

def main():
    p = argparse.ArgumentParser()
    p.add_argument("-p", required=True)
    args = p.parse_args()

    api_key = os.getenv("OPENROUTER_API_KEY")
    base_url = os.getenv("OPENROUTER_BASE_URL")

    if not api_key:
        print("error: OPENROUTER_API_KEY is not set", file=sys.stderr)
        sys.exit(1)

    if not base_url:
        print("error: OPENROUTER_BASE_URL is not set", file=sys.stderr)
        sys.exit(1)

    client = OpenAI(api_key=api_key, base_url=base_url)
    chat = client.chat.completions.create(
        model="anthropic/claude-haiku-4.5",
        messages=[{"role": "user", "content": args.p}],
    )
    if not chat.choices:
        print("error: no choices in response", file=sys.stderr)
        sys.exit(1)
    content = chat.choices[0].message.content
    if not content:
        print("error: empty content in response", file=sys.stderr)
        sys.exit(1)
    print(content, end="")

if __name__ == "__main__":
    main()
